redash/query_runner/__init__.py

```diff
@@ -118,6 +119,23 @@ def _run_query_internal(self, query):
             raise Exception("Failed running query [%s]." % query)
         return json_loads(results)['rows']
 
+    def _get_table_sample(self, table_name):
+        if self.data_sample_query is None:
+            raise NotImplementedError()
+
+        query = self.data_sample_query.format(table=table_name)
+
+        results, error = self.run_query(query, None)
+        if error is not None:
+            raise Exception("Failed getting table sample.")
+
+        sample = {}
```

Comment:
Let's move the default sample dictionary down into the else clause of the `if len(rows) > 0` condition.